### DISCUSSION TITLE
rust analyzer don't work on flake base system?

### DISCUSSION THREAD

**Discussion by pio2398** [srid/rust-nix-template#30] [General]
It seam that Rust Analyzer still trying use something outside of flake. My system don't have channels and extension crash.

I am the only one with this problem?
 
```
INFO [21/01/2024, 12:14:12]: Starting language client
ERROR [21/01/2024, 12:14:14]: Bootstrap error [Error: warning: Nix search path entry '/nix/var/nix/profiles/per-user/root/channels/nixos' does not exist, ignoring
error:
       … while calling the 'import' builtin

         at «stdin»:2:29:

            1|
            2|             {srcStr, pkgs ? import <nixpkgs> {}}:
             |                             ^
            3|                 pkgs.stdenv.mkDerivation {

       … while calling the 'findFile' builtin

         at «stdin»:2:36:

            1|
            2|             {srcStr, pkgs ? import <nixpkgs> {}}:
             |                                    ^
            3|                 pkgs.stdenv.mkDerivation {

       error: file 'nixpkgs' was not found in the Nix search path (add it using $NIX_PATH or -I)

       at «none»:0: (source not available)

	at /home/piotrek/.vscode-oss/extensions/rust-lang.rust-analyzer-0.4.1813-linux-x64/out/main.js:354:291
	at ChildProcess.exithandler (node:child_process:438:5)
	at ChildProcess.emit (node:events:513:28)
	at maybeClose (node:internal/child_process:1091:16)
	at ChildProcess._handle.onexit (node:internal/child_process:302:5)]

```

**Comment by srid** [comment]
Have you tried running it under `nix develop`?

Note: we have a [Zulip](https://nixos.zulipchat.com/#narrow/stream/413950-nix) for chat discussions.

**Reply by srid** [reply]
If you are not familiar with `nix *` commands, see:

- https://nixos.asia/en/nix-first
- https://nixos.asia/en/nix-rapid

**Reply by pio2398** [reply]
You mean staring VsCode in nix develop shell? That don't solve problem. As I understand this is direnv task anyway.

I think that Rust Analyzer VsCode extension need channel on base operating system.
Thanks for the links, but since my whole system is based on flake I won't learn much new from them.

**Reply by srid** [reply]
As I remember, this repo worked on VSCode fine with direnv on NixOS. I did setup direnv using https://github.com/juspay/nix-dev-home

**Reply by pio2398** [reply]
I can confirm that rust-analyzer don't support system without channel. Problem is here:  https://github.com/rust-lang/rust-analyzer/blob/47b0f466f5442cd9d124c47421f490b5a2f29c52/editors/code/src/bootstrap.ts#L109
nixpkgs should be used from project flake for best results.

I added channel and everything seem to work. Upstream issue but since flakes are experimental probably I won't report problem.

Thanks.

**Comment by pio2398** [comment]
You mean staring VsCode in nix develop shell? That don't solve problem.

I think that Rust Analyzer VsCode extension need channel on base operating system.
Thanks for the links, but since my whole system is based on flake I won't learn much new from them.
